engines/scumm/gfx_gui.cpp

```diff
@@ -2559,6 +2748,217 @@ void ScummEngine::setUpMainMenuControls() {
+							getBannerColor(20),
+							getBannerColor(6),
+							getBannerColor(7),
+							isJap ? 26 : 44,
+							yConstant - 5,
+							isJap ? 111 : 103,
+							yConstant - 5 + 22,
+							getGUIString(gsPlay), 1, 1);
+
+	// Restart button
+	setUpInternalGUIControl(GUI_CTRL_RESTART_BUTTON,
+							getBannerColor(4),
+							getBannerColor(5),
+							getBannerColor(17),
+							getBannerColor(18),
+							getBannerColor(19),
+							getBannerColor(20),
+							getBannerColor(6),
+							getBannerColor(7),
+							isJap ? 26 : 44,
+							yConstant + 21,
+							isJap ? 111 : 103,
+							yConstant + 21 + 22,
+							getGUIString(gsRestart), 1, 1);
+
+	if (_menuPage == GUI_PAGE_MAIN) {
+		// Arrow left button
+		setUpInternalGUIControl(GUI_CTRL_ARROW_LEFT_BUTTON,
+								getBannerColor(4),
+								getBannerColor(5),
+								getBannerColor(17),
+								getBannerColor(18),
+								getBannerColor(19),
+								getBannerColor(20),
+								getBannerColor(6),
+								getBannerColor(7),
+								235,
+								yConstant + 18,
+								247,
+								yConstant + 34,
+								_arrowLeft, 1, 1);
+
+		// Arrow right button
+		setUpInternalGUIControl(GUI_CTRL_ARROW_RIGHT_BUTTON,
+								getBannerColor(4),
+								getBannerColor(5),
+								getBannerColor(17),
+								getBannerColor(18),
+								getBannerColor(19),
+								getBannerColor(20),
+								getBannerColor(6),
+								getBannerColor(7),
+								248,
+								yConstant + 18,
+								260,
+								yConstant + 34,
+								_arrowRight, 1, 1);
+
+		setUpInternalGUIControl(GUI_CTRL_TEXT_SPEED_SLIDER,
+								getBannerColor(9),
+								getBannerColor(10),
+								getBannerColor(18),
+								getBannerColor(17),
+								getBannerColor(20),
+								getBannerColor(19),
+								getBannerColor(10),
+								getBannerColor(12),
+								152,
+								yConstant + 18,
+								235,
+								yConstant + 34,
+								_uncheckedBox, 1, 1);
+	}
+
+	if (_menuPage == GUI_PAGE_RESTART || _menuPage == GUI_PAGE_CODE_CONFIRM) {
+		// OK button
+		setUpInternalGUIControl(GUI_CTRL_OK_BUTTON,
+								getBannerColor(4),
+								getBannerColor(5),
+								getBannerColor(17),
+								getBannerColor(18),
+								getBannerColor(19),
+								getBannerColor(20),
+								getBannerColor(6),
+								getBannerColor(7),
+								isJap ? 115 : 149,
+								yConstant + 21,
+								200,
+								yConstant + 43,
+								getGUIString(gsOK), 1, 1);
+
+		// Cancel button
+		setUpInternalGUIControl(GUI_CTRL_CANCEL_BUTTON,
+								getBannerColor(4),
+								getBannerColor(5),
+								getBannerColor(17),
+								getBannerColor(18),
+								getBannerColor(19),
+								getBannerColor(20),
+								getBannerColor(6),
+								getBannerColor(7),
+								206,
+								yConstant + 21,
+								isJap ? 291 : 257,
+								yConstant + 43,
+								getGUIString(gsCancel), 1, 1);
+	}
+
+	if (_menuPage == GUI_PAGE_LOAD) {
+		Common::String numbers[] = {"0", "1", "2", "3", "4", "5", "6", "7", "8", "9"};
```

Comment:
`const char *numbers[2] = ....`

Same effect but skips the instantiation of 10 objects

Reply:
Thanks!